src/simulations/collect.jl

```diff
@@ -8,6 +8,16 @@ export run!, collect_agent_data!, collect_model_data!,
 get_data(a, s::Symbol, obtainer::Function) = obtainer(getproperty(a, s))
 get_data(a, f::Function, obtainer::Function) = obtainer(f(a))
 
+get_data_missing(a, s::Symbol, obtainer::Function) =
+    hasproperty(a, s) ? obtainer(getproperty(a, s)) : missing
+function get_data_missing(a, f::Function, obtainer::Function)
+    try
+        obtainer(f(a))
+    catch
+        missing
+    end
```

Comment:
I didn't have enough time to review this PR normally, but here I need to leave the comment that `try` blocks lead to horrible performance and should not be used in performance-critical code. 

The question is two fold. Is this performance critical code, and is there a non-try way to do it?

Reply:
1. It's for collections on mixed models that already provide a performance warning, so that's already covered.
2. This is for checking that arbitrary user functions that operate over an arbitrary collection of possible agent types doesn't spit an error. If this isn't the place for a try/catch, then there is no place :smile:

Reply:
In addition: its only ever called in the blocks with `T>:Missing`, so it's essentially only for times that users are fast and loose with sanitation of their data. The docs suggest not doing this.